pkg/ctl/install/cmd.go

```diff
@@ -0,0 +1,100 @@
+package install
+
+import (
+	"context"
+	"time"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+
+	api "github.com/weaveworks/eksctl/pkg/apis/eksctl.io/v1alpha5"
+	"github.com/weaveworks/eksctl/pkg/ctl/cmdutils"
+)
+
+// Command will create the `install` commands
+func Command(flagGrouping *cmdutils.FlagGrouping) *cobra.Command {
+	verbCmd := cmdutils.NewVerbCmd("install", "Install components in a cluster", "")
+
+	cmdutils.AddResourceCmd(flagGrouping, verbCmd, installFluxCmd)
+
+	return verbCmd
+}
+
+type installFluxOpts struct {
+	gitURL              string
+	gitBranch           string
+	gitPaths            []string
+	gitLabel            string
+	gitUser             string
+	gitEmail            string
+	gitFluxPath         string
+	namespace           string
+	tillerNamespace     string
+	timeout             time.Duration
+	amend               bool
+	noHelmOp            bool
+	noTiller            bool
+	tillerHost          string
+	helmOpTLSCertFile   string
+	helmOpTLSKeyFile    string
+	helmOpTLSCACertFile string
+}
+
+func installFluxCmd(cmd *cmdutils.Cmd) {
+	cmd.ClusterConfig = api.NewClusterConfig()
+	cmd.SetDescription(
+		"flux",
+		"Bootstrap Flux, installing it in the cluster and initializing its manifests in the specified Git repository",
+		"",
+	)
+	var opts installFluxOpts
+	cmd.SetRunFuncWithNameArg(func() error {
+		installer, err := newFluxInstaller(context.Background(), cmd, &opts)
+		if err != nil {
+			return err
+		}
+		return installer.run(context.Background())
+	})
+	cmd.FlagSetGroup.InFlagSet("Flux installation", func(fs *pflag.FlagSet) {
+		fs.StringVar(&opts.gitURL, "git-url", "",
+			"URL of the Git repository to be used by Flux, e.g. git@github.com:<your username>/flux-get-started")
+		fs.StringVar(&opts.gitBranch, "git-branch", "master",
+			"Git branch to be used by Flux")
+		fs.StringSliceVar(&opts.gitPaths, "git-paths", []string{},
+			"Relative paths within the Git repo for Flux to locate Kubernetes manifests")
+		fs.StringVar(&opts.gitLabel, "git-label", "flux",
+			"Git label to keep track of Flux's sync progress; overrides both --git-sync-tag and --git-notes-ref")
+		fs.StringVar(&opts.gitUser, "git-user", "Flux",
+			"Username to use as Git committer")
+		fs.StringVar(&opts.gitEmail, "git-email", "",
+			"Email to use as Git committer")
+		fs.StringVar(&opts.gitFluxPath, "git-flux-subdir", "flux/",
```

Comment:
Doesn't seem like an essential flag to expose.

Reply:
Unrelated to this PR (this was done in #1064 )

Reply:
Sure, I am just saying that there are many non-essential flags. We should expose flags for theoretical cases where user may need an option. If user needs many options, they should just use a different way of installing these components, there are several to chose from.

Reply:
Right, please raise a ticket. Let's please stick to reviewing the code added in the PR.